lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/GenericMathLangAST.hs

```diff
@@ -122,22 +122,22 @@ makeFieldLabelsNoPrefix ''ExplnAnnot
 --
 -- (Row and column, but not filename.)
 --
-data SrcPositn = MkPositn
+data SrcPosition = MkPosition
   { row :: !Int
   , col :: !Int
   } deriving stock (Eq, Ord, Show, Generic)
     deriving Hashable
-makeFieldLabelsNoPrefix ''SrcPositn
+makeFieldLabelsNoPrefix ''SrcPosition
 
 -- | Metadata stored with an expression.
 --
 data ExpMetadata =
   MkExpMetadata
-  { srcPos     :: SrcPositn        -- ^ source position of annotated expression
+  { srcPos     :: SrcPosition        -- ^ source position of annotated expression
   , typeLabel  :: Maybe TLabel     -- ^ (possibly) type information for annotated expression
   , explnAnnot :: Maybe ExplnAnnot -- ^ (possibly) additional annotations
   }
-  deriving stock (Eq, Show)
+  deriving stock (Eq, Ord, Show)
```

Comment:
Ah I see where this question about `Ord` is coming from now.

I *may* have been the one who hadn't put `Ord` in at the start. If so, there probably wasn't a good reason for this --- it was probably just that (i) I hadn't needed it at the time and (ii) I can sometimes be a bit of a micro-optimizer, and I probably hadn't wanted to add things that might unnecessarily increase compile times.

In retrospect, I doubt that deriving `Ord` would really increase compile times by very much at all.

Reply:
Indeed, but it is not like I am using `Ord` for anything, I was just curious, whether the code doesn't permit it for any fundamental reason.

I think there a good reasons to avoid deriving unnecessarily. Especially, if you don't actually want people to use it, as comparing trees can be inefficient. GHC, for example, also avoids deriving `Ord`.